mev_inspect/jit_liquidity.py

```diff
@@ -0,0 +1,233 @@
+        ):
+            mint_slice_end = index
+        if (
+            burn_slice_start is None
+            and trace.transaction_hash == burn_trace.transaction_hash
+        ):
+            burn_slice_start = index
+        if (
+            burn_slice_end is None
+            and trace.transaction_position > burn_trace.transaction_position
+        ):
+            burn_slice_end = index
+            break
+
+    mint_net_transfers_full = get_net_transfers(
+        classified_traces[mint_slice_start:mint_slice_end]
+    )
+    burn_net_transfers_full = get_net_transfers(
+        classified_traces[burn_slice_start:burn_slice_end]
+    )
+
+    mint_net_transfers, burn_net_transfers = [], []
+    pool_address = mint_trace.to_address
+
+    for transfer in mint_net_transfers_full:
+        if transfer.to_address == pool_address:
+            mint_net_transfers.append(transfer)
+
+    for transfer in burn_net_transfers_full:
+        if transfer.from_address == pool_address:
+            burn_net_transfers.append(transfer)
+
+    if len(mint_net_transfers) > 2 or len(burn_net_transfers) > 2:
+        error_found = True
+
+    token0_address, token1_address = _get_token_order(
+        mint_net_transfers[0].token_address, mint_net_transfers[1].token_address
+    )
+    if mint_net_transfers[0].token_address == token0_address:
+        mint_token0 = mint_net_transfers[0].amount
+        mint_token1 = mint_net_transfers[1].amount
+    else:
+        mint_token0 = mint_net_transfers[1].amount
+        mint_token1 = mint_net_transfers[0].amount
+
+    if burn_net_transfers[0].token_address == token0_address:
+        burn_token0 = burn_net_transfers[0].amount
+        burn_token1 = burn_net_transfers[1].amount
+    else:
+        burn_token0 = burn_net_transfers[1].amount
+        burn_token1 = burn_net_transfers[0].amount
+
+    return JITTransferInfo(
+        token0_address=token0_address,
+        token1_address=token1_address,
+        mint_token0=mint_token0,
+        mint_token1=mint_token1,
+        burn_token0=burn_token0,
+        burn_token1=burn_token1,
+        error=error_found,
+    )
+
+
+def _get_bot_address(
+    mint_trace: ClassifiedTrace, classified_traces: List[ClassifiedTrace]
+) -> str:
+    if mint_trace.from_address in LIQUIDITY_MINT_ROUTERS:
+        bot_trace = list(
+            filter(
+                lambda t: t.to_address == mint_trace.from_address
+                and t.transaction_hash == mint_trace.transaction_hash,
+                classified_traces,
+            )
+        )
+        if len(bot_trace) == 1 or is_child_trace_address(
+            bot_trace[1].trace_address, bot_trace[0].trace_address
+        ):
+            return _get_bot_address(bot_trace[0], classified_traces)
+        else:
+            return "0x0000000000000000000000000000000000000000"
+
+    # This case is here because from_address is optional in ClassifiedTrace
+    if type(mint_trace.from_address) == str:
```

Comment:
If an attribute (from_address) is Optional in the schema, the object can be created without passing a value and the key will not exist in the dictionary. This will then throw a KeyError. I believe a better check would be 
```if 'from_address' in mint_trace.keys()```